documentation/modules/oauth/con-oauth-authentication-broker.adoc

```diff
@@ -0,0 +1,135 @@
+// Module included in the following assemblies:
+//
+// assembly-oauth.adoc
+
+[id='con-oauth-authentication-broker-{context}']
+= {oauth} Kafka broker configuration
+
+Kafka broker configuration for {oauth} involves:
+
+* Client configuration (for the Kafka broker) on an authorization server
+* {oauth} authentication configuration in the Kafka cluster
+
+NOTE: In relation to the authorization server, Kafka brokers and Kafka clients are both regarded as {oauth} clients.
+
+== {oauth} client configuration on an authorization server
+
+To configure a Kafka broker to validate the token received during session initiation,
+the recommended approach is to create a {oauth} _client_ definition in an authorization server, configured as _confidential_, with client credentials enabled:
+
+* Client ID of `Kafka`
+* Client ID and secret as the authentication mechanism
+
+== {oauth} authentication configuration in the Kafka cluster
+
+To use {oauth} authentication in the Kafka cluster, you specify a TLS listener configuration for your Kafka cluster custom resource with the authentication method `oauth`:
+
+.Authentication method type for {oauth}
+[source,yaml,subs="+quotes, attributes"]
+----
+apiVersion: {KafkaApiVersion}
+kind: Kafka
+spec:
+  kafka:
+    listeners:
+      *tls*:
+        authentication:
+          type: *oauth*
+          #...
+----
+
+You can configure `plain`, `tls` and `external` listeners, as described in xref:assembly-configuring-kafka-broker-listeners-deployment-configuration-kafka[Kafka broker listeners],
+but it is recommended not to use `plain` listeners with {oauth} as this creates a vulnerability to network eavesdropping and unauthorized access through token theft.
+
+You configure an `external` listener with `tls: true` for a secure transport layer to communicate with the client.
+
+.External listener with TLS
+[source,yaml,subs="+quotes"]
+----
+# ...
+listeners:
+  tls:
+    authentication:
+      type: oauth
+  external:
+    type: oauth
+    *tls: true*
+    authentication:
+      type: tls
+# ...      #...
+----
+
+== Fast local JWT token validation configuration
+
+Fast local JWT token validation checks a JWT token signature locally.
+
+The local check ensures a token:
+
+* Conforms to type (`oauth`)
```

Comment:
Should the type here be JWT token instead of `oauth`?

Reply:
@mstruk - do we have the correct type here for Fast local JWT token validation configuration?

Reply:
It should be `Bearer`.

Reply:
To clarify, a JWT token 'conforms to type' by containing a 'typ' claim with value equal to 'Bearer'. It's on of the checks performed by default. The value 'Bearer' means that the token is an access token rather than a refresh token for example.

Reply:
@mstruk - thanks for clarifying. I've updated to: Conforms to type by containing a (_typ_) claim value of `Bearer`